source/dofs/dof_accessor_get.cc

```diff
@@ -69,15 +69,23 @@ DoFCellAccessor<DoFHandlerType, lda>::get_interpolated_dof_values(
           // well, here we need to first get the values from the current
           // cell and then interpolate it to the element requested. this
           // can clearly only happen for hp::DoFHandler objects
-          Vector<number> tmp(this->get_fe().dofs_per_cell);
-          this->get_dof_values(values, tmp);
-
-          FullMatrix<double> interpolation(
-            this->dof_handler->get_fe(fe_index).dofs_per_cell,
-            this->get_fe().dofs_per_cell);
-          this->dof_handler->get_fe(fe_index).get_interpolation_matrix(
-            this->get_fe(), interpolation);
-          interpolation.vmult(interpolated_values, tmp);
+          const unsigned int dofs_per_cell = this->get_fe().dofs_per_cell;
+          if (dofs_per_cell == 0)
+            {
+              interpolated_values = 0;
```

Comment:
@masterleinad : I'm way late to this game, but isn't it the case that if you end in this branch, `interpolated_values` needs to have size zero anyway? If so, one could just skip the `if` and jump straight to the `else`. If you agree, I can write that patch.

Reply:
No, in the test for this pull request the size of `interpolated_values` is 4 since the new element is a `FE_Q<2>(1)`.

Reply:
Oh, I see, it's the other way around. Thanks for confirming!